### PR TITLE
Don't extend FileIO's load and save

### LINKED ISSUE #19: segfault when passing an IOStream to JLD2.load
After
```
a = sprand(100, 100, 0.5)
@save "testjld2" a
```

I tried to pass an IOStream to `load`:
```
f = open("testjld2", "r")
# IOStream(<file testjld2>)

JLD2.load(f)
Segmentation fault: 11
```

~It'd be great if JLD2.load could take an IOStream.~ The docs imply that it should work.

### PR DESCRIPTION
**Pull request by timholy**
Fixes #19

### CODE REVIEW

**Comment by timholy** [comment]
Obviously the negative here is that users now have to remember to say `using FileIO`.

File: README.md [hunk 1]
```diff
@@ -15,6 +15,7 @@ The code here should work on Julia 0.6. It has extensive unit tests, but it has
 The `@save` and `@load` macros are the simplest way to interact with a JLD2 file. The `@save` macro writes one or more variables from the current scope to the JLD file. For example:
 
 ```julia
+using JLD2, FileIO
```
Comment (simonster) [inline README.md]: I don't think `FileIO` is necessary here, since the macro form is still provided by JLD2.

**Comment by simonster** [comment]
Thanks. I just copied this code from JLD; I didn't realize that these weren't supposed to be extending FileIO.

File: README.md (hunk 1 above)
Comment (timholy) [inline README.md]: True. I was just using it as another reminder, but if you prefer I'll get rid of it.

**Pull request merged**